Jellyfin.Plugin.AniDB/Providers/equals_check.cs

```diff
@@ -20,528 +18,217 @@ public Equals_check(ILogger<Equals_check> logger)
-            if (a.Contains("gekijyouban"))
-               a = (a.Replace("gekijyouban", "") + " Movie").Trim();
-            try
-            {
-                a = a.Trim().Replace(await One_line_regex(new Regex(@"(?s) \(.*?\)"), a.Trim(), cancellationToken, 0), "");
-            }
-            catch (Exception)
-            { }
-            a = a.Replace(".", " ");
-            a = a.Replace("-", " ");
-            a = a.Replace("`", "");
-            a = a.Replace("'", "");
-            a = a.Replace("&", "and");
-            a = a.Replace(":", "");
-            a = a.Replace("␣", "");
-            a = a.Replace("2wei", "zwei");
-            a = a.Replace("3rei", "drei");
-            a = a.Replace("4ier", "vier");
-            return a;
-        }
-
-        /// <summary>
-        /// If a and b match it return true
-        /// </summary>
-        /// <param name="a"></param>
-        /// <param name="b"></param>
-        /// <returns></returns>
-        public async static Task<bool> Compare_strings(string a, string b, CancellationToken cancellationToken)
-        {
-            if (!string.IsNullOrEmpty(a) && !string.IsNullOrEmpty(b))
-            {
-                if (await Simple_compare(a, b, cancellationToken))
-                    return true;
-                if (await Fast_xml_search(a, b, cancellationToken))
-                    return true;
-
-                return false;
-            }
-            return false;
-        }
-
         /// <summary>
         /// Cut p(%) away from the string
         /// </summary>
-        /// <param name="string_"></param>
-        /// <param name="min_lenght"></param>
+        /// <param name="input"></param>
+        /// <param name="minLenght"></param>
         /// <param name="p"></param>
         /// <returns></returns>
-        public async static Task<string> Half_string(string string_, CancellationToken cancellationToken, int min_lenght = 0, int p = 50)
+        public static string ShortenString(string input, int minLenght = 0, int p = 50)
         {
             decimal length = 0;
-            if (await Task.Run(() => ((int)((decimal)string_.Length - (((decimal)string_.Length / 100m) * (decimal)p)) > min_lenght), cancellationToken))
+            if ((int)((decimal)input.Length - (((decimal)input.Length / 100m) * (decimal)p)) > minLenght)
             {
-                length = (decimal)string_.Length - (((decimal)string_.Length / 100m) * (decimal)p);
+                length = (decimal)input.Length - (((decimal)input.Length / 100m) * (decimal)p);
             }
             else
             {
-                if (string_.Length < min_lenght)
+                if (input.Length < minLenght)
                 {
-                    length = string_.Length;
+                    length = input.Length;
                 }
                 else
                 {
-                    length = min_lenght;
+                    length = minLenght;
                 }
```

Comment:
I agree with GH CodeQL here, you could solve this with a ternary :wink:

Reply:
I didn't write that (just changed the variable name to camel case), but I agree and am going to change it.

Reply:
I changed the code for better performance and now I can't use a ternary anymore. There was no reason to use decimal instead of float, and to do the calculation twice. Also, it's better to just immediately return the input if it's shorter than minLength.